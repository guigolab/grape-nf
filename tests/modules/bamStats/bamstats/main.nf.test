nextflow_process {

    name "Test Process bamStats"
    script "modules/bamStats/bamstats/main.nf"
    process "bamStats"

    tag "module"

    test("Should produce BAM stats (compressed annotation)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf.gz")
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test("Should produce BAM stats (uncompressed annotation)") {

        when {
            process {
                """
                input[0] = file("${baseDir}/data/annotation.gtf")
                input[1] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
