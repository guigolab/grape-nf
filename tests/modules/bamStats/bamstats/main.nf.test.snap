{
    "Should produce BAM stats (uncompressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3_stats.json:md5,77c72bef6f1410e33c90dbc01697fffc",
                        "json",
                        "BamStats",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T17:10:13.589448"
    },
    "Should produce BAM stats (compressed annotation)": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3_stats.json:md5,77c72bef6f1410e33c90dbc01697fffc",
                        "json",
                        "BamStats",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-18T17:10:05.978606"
    }
}